datafusion/datasource/src/source.rs

```diff
@@ -35,7 +35,7 @@ use datafusion_physical_expr_common::sort_expr::LexOrdering;
 
 /// Common behaviors in Data Sources for both from Files and Memory.
 /// See `DataSourceExec` for physical plan implementation
-pub trait DataSource: Send + Sync {
+pub trait DataSource: Send + Sync + Debug {
```

Comment:
this means that anything that implements `DataSource` must also implement `Debug`